src/tools/rpc.rs

```diff
@@ -0,0 +1,224 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    time::Duration,
+};
+
+use data_encoding::{BASE32_NOPAD, BASE64};
+use fmt::Debug;
+use reqwest::{header, Client};
+use serde::{
+    de::{Error, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use tokio::time::{error::Elapsed, sleep};
+
+use crate::protocol::constants::USER_AGENT;
+
+/// Timeout time for RPC requests.
+const RPC_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// Returns a block for a provided round.
+pub async fn wait_for_block(rpc_addr: &str, round: u64) -> Result<EncodedBlockCert, Elapsed> {
+    // Algod V1 documentation states that the round format is 'integer (int64)',
+    // but it's actually an int64 integer encoded in base36.
+    let round = radix_fmt::radix_36(round).to_string();
+
+    tokio::time::timeout(RPC_TIMEOUT, async move {
+        loop {
+            if let Ok(rsp) = get_block(rpc_addr, &round).await {
```

Comment:
since this happens in a loop, it would likely be faster to make a `reqwest` `Client` persist across iterations